metabase/metabase
# Use RTK Query instead of entity framework for DatabaseCandidates

resolves https://github.com/metabase/metabase/issues/40792
### Description

A "database candidate" is a weird name for a suggested x-ray for a table. We suggest looking at these x-rays for new users in two places:

1. The database sync modal, which we show when an admin connects their first database

![Screen Shot 2024-03-20 at 12 53 44 PM](https://github.com/metabase/metabase/assets/30528226/57c70fff-56a0-42bc-8757-3b59baac9555)

2. The home screen, for new users.

![Screen Shot 2024-03-20 at 1 57 37 PM](https://github.com/metabase/metabase/assets/30528226/2fbc0025-4ad3-4f42-9958-ffc1a2c7b89b)

There's no reason this needs to be an entity. In fact it's much simpler and easier if it's "just" an api request.

### How to verify

1. Set up a fresh metabase instance and DO NOT connect a database during setup
2. After setup, go to admin> databases and connect a database. See that the database sync modal appears and links you to a table in the sample DB
3. Before viewing any questions, you should see suggested x-rays on the home page. If there are no connected DBs, these will be based on the sample database. If you connect a database, it'll be based on tables in there.

If you want to test again, you gotta reset your app db!

### Checklist
